File: ignored-test/java-junit/RenderShouldRecognizeStackTraceUsage.java
import org.junit.Test;
import static org.junit.Assert.*;

@Disabled
public class LayoutTypedTests {

    @Test
    public void renderShouldRecognizeStackTraceUsage() {
        // Arrange
        Object[] callbackArgs = null;
        BiConsumer<LogEventInfo, Object[]> callback = (evt, args) -> callbackArgs = args;
        Logger logger = new LogFactory().setup().loadConfiguration(builder -> {
            MethodCallTarget methodCall = new MethodCallTarget("dbg", callback);
            methodCall.getParameters().add(new MethodCallParameter("LineNumber", "${callsite-linenumber}", Integer.class));
            builder.forLogger().writeTo(methodCall);
        }).getLogger("renderShouldRecognizeStackTraceUsage");
        // Act
        logger.info("Testing");
        // Assert
        assertEquals(1, callbackArgs.length);
        int lineNumber = (Integer) callbackArgs[0];
        assertTrue(lineNumber > 0);
    }
}

File: refactored/ignored-test/CSharpXUnit/CSharpXUnit.Tests/Tests/RenderShouldRecognizeStackTraceUsage.cs
using Xunit;

public class LayoutTypedTests
{
    [Fact]
    // [Fact(Skip = "Disabled")]
    public void RenderShouldRecognizeStackTraceUsage()
    {
        // Arrange
        object[] callbackArgs = null;
        Action<LogEventInfo, object[]> callback = (evt, args) => callbackArgs = args;
        Logger logger = new LogFactory().Setup().LoadConfiguration(builder => {
            MethodCallTarget methodCall = new MethodCallTarget("dbg", callback);
            methodCall.Parameters.Add(new MethodCallParameter("LineNumber", "${callsite-linenumber}", typeof(int)));
            builder.ForLogger().WriteTo(methodCall);
        }).GetLogger("RenderShouldRecognizeStackTraceUsage");
        
        // Act
        logger.Info("Testing");
        
        // Assert
        Assert.Equal(1, callbackArgs.Length);
        int lineNumber = (int)callbackArgs[0];
        Assert.True(lineNumber > 0);
    }
}
